src/app/JoinServer.js

```diff
@@ -91,14 +88,29 @@ class JoinServer {
 		this.routes()
 	}
 
+	newDataUnionClient(chain, privateKey) {
+		const chains = config.Chains.load()
+		const options = {
+			auth: {
+				privateKey,
+			},
+			network: {
+				name: chains[chain].name,
+				chainId: chains[chain].id,
+				rpcs: chains[chain].rpcEndpoints,
+			}
+		}
+		return new DataUnionClient(options)
+	}
+
 	routes() {
 		this.expressApp.use(express.json({
 			limit: '1kb',
 		}))
-		this.expressApp.use((req, res, next) => this.signedRequestValidator(req).then(next).catch((err) => next(err)))
+		this.expressApp.use('/join', (req, _res, next) => this.signedRequestValidator(req).then(next).catch((err) => next(err)))
```

Comment:
The middleware should apply to all routes, including ones added by `customRoutes` - so the original code should be reverted